common/utils.js

```diff
@@ -5,19 +5,105 @@
 
     .factory('UriUtils', ['$injector', '$window', function($injector, $window) {
 
-        return {
+        /**
+        * @function
+        * @param {String} str string to be encoded.
+        * @desc
+        * converts a string to an URI encoded string
+        */
+        function fixedEncodeURIComponent(str) {
+            return encodeURIComponent(str).replace(/[!'()*]/g, function (c) {
+                return '%' + c.charCodeAt(0).toString(16).toUpperCase();
+            })
+        }
+
+        function parseURLFragment(context) {
```

Comment:
I'm okay with this interface. But if I were designing this as an api, I would have made it:
```
function parseURLFragment(location, context) 
```
And not have it dependent on $window directly. I would allow `context` to be `null` and if the caller didn't pass that parameter create a new `context` object. And then the function would always return `context` (whether a new instance or the passed in instance). Caller could use it like:
```
var context = UriUtils.parseURLFragment($window.location);
```
or
```
/* I already have a mycontext */
UriUtils.parseURLFragment($window.location, mycontext);
```